packages/analytics/src/Analytics.ts

```diff
@@ -406,7 +406,11 @@ class Analytics {
     try {
       this.userInstance && this.userInstance.anonymize();
 
-      this.handleOnUserChanged();
+      await this.handleOnUserChanged();
+
+      this.setUserPromise = new Promise(resolve => {
```

Comment:
I think this line should be set before calling `handleOnUserChanged` method because of the await it might take a while to come back and execute the change of the `setUserPromise` and while that is going on, another track can happen.